### PR TITLE
useAPIPagination + useTxs

### PR DESCRIPTION
**Pull request by marinhoarthur**
Mind:
- useAPIPagination
- useTxs
- SpTxList (just a demo)

### CODE REVIEW

**Comment by clockworkgr** [comment]
fwiw, autogenerated vuex module actions for querying accept query params to handle paging 

e.g.`query` here: https://github.com/tendermint/vue/blob/develop/packages/template/src/store/generated/cosmos/cosmos-sdk/cosmos.bank.v1beta1/index.ts#L181

assuming you use `all: false`

You can always use `all: true` which will loop through next_keys and load everything into the store:

https://github.com/tendermint/vue/blob/develop/packages/template/src/store/generated/cosmos/cosmos-sdk/cosmos.bank.v1beta1/index.ts#L188-L191

**Comment by marinhoarthur** [comment]
> fwiw, autogenerated vuex module actions for querying accept query params to handle paging
> 
> e.g.`query` here: https://github.com/tendermint/vue/blob/develop/packages/template/src/store/generated/cosmos/cosmos-sdk/cosmos.bank.v1beta1/index.ts#L181
> 
> assuming you use `all: false`
> 
> You can always use `all: true` which will loop through next_keys and load everything into the store:
> 
> https://github.com/tendermint/vue/blob/develop/packages/template/src/store/generated/cosmos/cosmos-sdk/cosmos.bank.v1beta1/index.ts#L188-L191

1. `next_key` might not be present as in [here](https://github.com/cosmos/cosmos-sdk/issues/10511) 
2.  imo vuex is not the place to have this kind of logic 
3.  how can one paginate to a specific txs "page" using current code?

File: packages/template/src/App.vue [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
+  <div>
     <SpTheme>
       <SpNavbar
         :links="navbarLinks"
         :activeRoute="router.currentRoute.value.path"
       />
       <router-view />
+      <Suspense>
+        <SpTxList v-if="address" :address="address" />
+      </Suspense>
```
Comment (mihailmarcu) [inline packages/template/src/App.vue]: I guess the `<suspense>` component was not used correctly here. If you want to use it use the `<template #default>` slot in this case.

File: packages/vue/src/components/SpTxList/SpTxList.vue [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    return {
+      receivedTxsPager,
+      sentTxsPager
+    }
+  }
+})
+</script>
+
+<style lang="scss">
```
Comment (mihailmarcu) [inline packages/vue/src/components/SpTxList/SpTxList.vue]: Let's use the `scoped` attribute here to exclude some conflicts later.

**Review by mihailmarcu — APPROVED** [review]
Hey @marinhoarthur,
Overall the PR looks good and could be merged. I noticed only 2 minor things that could be corrected or postponed.

**Pull request merged**